lib/blacklight/solr_helper.rb

```diff
@@ -78,7 +78,17 @@ def solr_doc_params(id=nil)
   # Returns a two-element array (aka duple) with first the solr response object,
   # and second an array of SolrDocuments representing the response.docs
   def get_search_results(user_params = params || {}, extra_controller_params = {})
-    solr_response = query_solr(user_params, extra_controller_params)
+    Deprecation.warn(self, "get_search_results is deprecated and will be removed in blacklight-6.0. Use `search_results' instead")
+    search_results(user_params, extra_controller_params, solr_search_params_logic)
+  end
+
+  # a solr query method
+  # @param [Hash,HashWithIndifferentAccess] user_params ({}) the user provided parameters (e.g. query, facets, sort, etc)
+  # @param [Hash,HashWithIndifferentAccess] extra_controller_params ({}) extra parameters to add to the search
+  # @param [List<Symbol] processor_chain a list of filter methods to run
+  # @return [Blacklight::SolrResponse] the solr response object
+  def search_results(user_params, extra_controller_params, solr_search_parms_logic)
```

Comment:
While we're changing this, I wonder if we can do away with `extra_controller_params`.. Maybe a new type of `solr_search_params_logic` that appends the attributes?